misc/LADYBUG_USER_STORIES.md

```diff
@@ -3,51 +3,152 @@ Ladybug User Stories
+* Thread start point.
+* Thread end point.
+
+**1020:** As a Frank tester, I want to see a start point as the parent node of the checkpoints that come after it. This applies recursively: a start node inside a start node causes subsequent nodes to be grand children of the first start node. An end point that is a direct child of a start node is also the last child. Subsequent nodes are siblings of the ended start node. This also applies recursively.
+
+**1030:** As a Frank tester I want to be able to collapse and to expand each parent node in the tree view. Each parent node can be *expanded* which means that the start node and its children, including the end node, are shown. A parent node can also be *collapsed* which means that its descendants are not shown. This way I can hide details of how a message was processed.
+
+**1033:** As a Frank tester, I want to have a checkpoint for each pipe such that I can see how each pipe transformed the incoming message.
+
+**1037:** As a Frank tester, I want a single report when my adapter calls another adapter via a JavaListener. This way, I do not have to browse multiple reports to examine how my incoming message was processed. The sub-adapter checkpoints should have a common ancestor. When I collapse that ancestor, I want to see only one node for everything done in the sub-adapter.
+
+**1040:** As a Frank tester, I want each report to have a single parent node. That parent node does not correspond to a checkpoint. All other nodes correspond to checkpoints and they are descendants of the parent node. This helps me to see which checkpoint belongs to which report. The name of the parent node is the same as the name of the first checkpoint, which is the outer start point.
+
+**1050:** As a Frank tester, I want to see the message of a checkpoint when I click on it.
+
+TODO: Is there more information than the message here?
+
+**1100:** As a Frank tester, I want the option to remove a report from the tree view when I am done with it. This does not mean that a report is removed from persistent storage.
+
+# I want to turn a report into a test case
+
+**2000:** Given is that I am viewing a report as a Frank tester. I want the option to edit the message within each checkpoint. When the report is rerun, the produced message will be compared to the edited message instead of the message originally captured. This allows me to update reports as test cases when my Frank application is changed.
+
+**2010:** Given is that I am viewing a report as a Frank tester. I want the option to configure an XSLT transformation that is applied to each message inside each checkpoint. When a report is rerun, the XSLT transformation is applied to the produced messages and it is applied to the messages inside the checkpoints. For each checkpoint, the two transformation results are compared. This way, irrelevant differences can be ignored. Irrelevant differences are produced for example if the current time is used by a Frank application.
```

Comment:
dont get this one

Reply:
I am describing how we do ignores in Ladybug. I describe here how XSLT is employed for this.